include/seqan3/io/sequence/sequence_file_in.hpp

```diff
@@ -1,53 +1,113 @@
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Svenja, Temesgen, Jongkyu <our_emails@fu-berlin.de>
+// ==========================================================================
+// sequence_file_in.hpp
+// ==========================================================================
+
 #pragma once
 
+#include <fstream>
 #include <string>
 #include <variant>
 #include <vector>
+#include <functional>
+#include "sequence_file_format.hpp"
+#include "sequence_file_format_fasta.hpp"
 
 namespace seqan3
 {
 
 // ==================================================================
 // sequence_file_in_traits
 // ==================================================================
-
 template <typename t>
 concept bool sequence_file_in_traits_concept = requires (t v)
 {
-    t::stream_type;
-    t::valid_formats;
+    typename t::stream_type;
+    typename t::valid_formats;
+    // //TODO valid_formats shall be variant and all of variants types
+    // // must meet format_concept
     requires detail::meets_concept_sequence_file_format<0, typename t::valid_formats>();
 
     t::valid_compression_formats;
 };
 
 struct sequence_file_in_default_traits
 {
-    using stream_type = std::ofstream;
-    using valid_formats = std::variant<sequence_file_format_fasta<stream_type>,
-                                       sequence_file_format_fastq<stream_type>,
-                                       sequence_file_format_embl<stream_type>,
-                                       sequence_file_format_genbank<stream_type>,
-                                       sequence_file_format_raw<stream_type>>;
-    static constexpr std::vector<std::pair<std::string, void>> valid_compression_formats{};
+    using stream_type = std::ifstream;
+    using valid_formats = std::variant<sequence_file_format_fasta>;
+                                       // sequence_file_format_fastq,
+                                       // sequence_file_format_embl,
+                                       // sequence_file_format_genbank,
+                                       // sequence_file_format_raw>;
+
+    using valid_compressions = std::variant<char/*specify compression formats*/>;
+
+    static inline std::vector<std::pair<std::string, valid_compressions>> valid_compression_formats
+    {
+    //     {".gz", variant_alternative_t<0, valid_compressions>{}},
+    //     {".bz2", variant_alternative_t<1, valid_compressions>{}}
+    };
 };
 
+static_assert(sequence_file_in_traits_concept<sequence_file_in_default_traits>);
```

Comment:
This does not need to be checked here. This can be added to the test file for the traits object.
Otherwise, the sequence_file_in_traits already checks the concept.